src/utils/lessonStates/index.js

```diff
@@ -27,157 +27,101 @@ import {
   retireActionText,
 } from 'utils/text'
 
-const lessonStates = [
-  'proposed',
-  'cancelled',
-  'accepted',
-  'requested',
-  'claimed',
-  'submitted',
-  'rejected',
-  'updated',
-  'approved',
-  'published',
-  'flagged',
-  'revised',
-  'retired',
-]
-
-export default lessonStates
-
-export const inProgressLessonStates = without(slice(
-  lessonStates,
-  indexOf(lessonStates, 'proposed'),
-  indexOf(lessonStates, 'approved')
-), 'cancelled')
-
-export const publishedLessonStates = slice(
-  lessonStates,
-  indexOf(lessonStates, 'published')
-)
-
-export const statusByLessonState = {
+export const detailsByLessonState = {
   proposed: {
+    action: cancelActionText,
     description: proposedStateDescriptionText,
+    color: 'white',
   },
   cancelled: {
-    requiresUserAction: true,
+    action: cancelActionText,
     description: cancelledStateDescriptionText,
+    color: 'red',
   },
   accepted: {
-    requiresUserAction: true,
+    action: acceptActionText,
     description: acceptedStateDescriptionText,
+    color: 'green',
   },
   requested: {
-    requiresUserAction: true,
+    action: requestActionText,
     description: requestedStateDescriptionText,
+    color: 'white',
   },
   claimed: {
-    requiresUserAction: true,
+    action: claimActionText,
     description: claimedStateDescriptionText,
+    color: 'green',
   },
   submitted: {
+    action: submitActionText,
     description: submittedStateDescriptionText,
+    color: 'green',
   },
   rejected: {
-    requiresUserAction: true,
+    action: rejectActionText,
     description: rejectedStateDescriptionText,
+    color: 'orange',
   },
   updated: {
+    action: applyUpdateActionText,
     description: updatedStateDescriptionText,
+    color: 'green',
   },
   approved: {
+    action: approveActionText,
     description: approvedStateDescriptionText,
+    color: 'green',
   },
   published: {
+    action: publishActionText,
     description: publishedStateDescriptionText,
+    color: 'green',
   },
   flagged: {
-    requiresUserAction: true,
+    action: flagActionText,
     description: flaggedStateDescriptionText,
+    color: 'orange',
   },
   revised: {
+    action: reviseActionText,
     description: revisedStateDescriptionText,
+    color: 'green',
   },
   retired: {
+    action: retireActionText,
     description: retiredStateDescriptionText,
+    color: 'red',
```

Comment:
I also used the new colors to update the lesson state badge to match:
![image](https://cloud.githubusercontent.com/assets/5497885/23692010/5d316866-0389-11e7-9b88-a860247680b9.png)